packages/block-library/src/gallery/index.php

```diff
@@ -32,12 +32,53 @@ function block_core_gallery_data_id_backcompatibility( $parsed_block ) {
 
 add_filter( 'render_block_data', 'block_core_gallery_data_id_backcompatibility' );
 
+/**
+ * Adds a style tag for the --wp--style--unstable-gallery-gap var.
+ *
+ * The Gallery block needs to recalculate Image block width based on
+ * the current gap setting in order to maintain the number of flex columns
+ * so a css var is added to allow this.
+ *
+ * @param array  $attributes Attributes of the block being rendered.
+ * @param string $content Content of the block being rendered.
+ * @return string The content of the block being rendered.
+ */
+function block_core_gallery_render( $attributes, $content ) {
+	$gap = _wp_array_get( $attributes, array( 'style', 'spacing', 'blockGap' ) );
+	// Skip if gap value contains unsupported characters.
+	// Regex for CSS value borrowed from `safecss_filter_attr`, and used here
+	// because we only want to match against the value, not the CSS attribute.
+	$gap       = preg_match( '%[\\\(&=}]|/\*%', $gap ) ? null : $gap;
+	$id        = uniqid();
+	$class     = 'wp-block-gallery-' . $id;
+	$content   = preg_replace(
+		'/' . preg_quote( 'class="', '/' ) . '/',
+		'class="' . $class . ' ',
+		$content,
+		1
+	);
+	$gap_value = $gap ? $gap : 'var( --wp--style--block-gap, 0.5em )';
```

Comment:
Should the fallback gap value be `16px` to be consistent with the previous / existing styling for classic themes?

Reply:
@andrewserong I have updated it to be 16px again, or `grid-unit-20` - it is a shame that the only way to do this requires some extra specificity and overriding the layout gap setting, even in block themes, would be nice to have a way of specifying the fallback in the layout support rather than just have 0.5em as the only option.

Reply:
> it would be nice to have a way of specifying the fallback in the layout support rather than just have 0.5em as the only option.

Apologies for the blow-in comment.

Isn't the fallback meant to be somewhat static, that is, where no block gap has been defined by the user or the theme? Given that, I'd tend to expect the fallback to be consistent across the app, just like `--wp--style--block-gap` is supposed to be (when defined). 

Not changing to `16px` will impact many existing galleries, which is tricky, but wouldn't it also create discrepancies between gallery and site-wide gap values in future galleries where no gap is defined at the block or theme level? I mean to say, the default layout gap value for the entire site is `0.5em` vs `16px` for galleries.

Personally, I'm not convinced that the 8 or so pixels difference is worth sacrificing the future consistency, especially since we're probably going to avoid using pixel values for future layouts. 

I can really come up with a decent backwards compatible alternative though 🤔 other than `if ( wp_is_block_theme() ) {`

Reply:
what do you think about @ramonjd's concerns @andrewserong? I don't have a strong opinion either way.

Reply:
I forgot to couch my comment in a _"This is a non-blocking opinion..."_ 😆

Reply:
I don't have too strong an opinion on it either — I think it's really a balancing act between backwards compatibility and how we think it should ideally work. I'm much more in favour personally of an `0.5em` fallback because the gap will be relative to the current font size, so (at least in theory) galleries should have a more pleasing default gap relative to the design.

Considering that moving to `gap` instead of margins is already a breaking change, I think it's fair to go with consistency here as @ramonjd's suggesting, rather than maintaining backwards compatibility at all costs.

> it is a shame that the only way to do this requires some extra specificity and overriding the layout gap setting, even in block themes

In short, I think that argument wins me over to keeping the `0.5em` and therefore not needing the override. Apologies for the back-and-forth here!

>  it would be nice to have a way of specifying the fallback in the layout support rather than just have 0.5em as the only option.

That's a great idea — perhaps one of the things we could look at in the layout work is seeing if we can define a default fallback value in `block.json` for the layout (or in the base `theme.json` if that's a better place for it) 🤔

Reply:
@andrewserong I have reverted that change now - no worries about the back and forth, was worth exploring.